tests/tests.js

```diff
@@ -0,0 +1,259 @@
+    assert.deepEqual(this.testmonth.dayIndex, {}, "Initialized the day Index object");
+    assert.equal(this.testmonth.monthId, "20171", "Initialized the month Id");
+});
+
+
+// Testing Calendar Helper functions
+QUnit.module( "Calendar helper functions tests", {
+});
+
+QUnit.test("generateUniqueId test", function(assert) {
+    assert.expect(1);
+    var uniqueId = generateUniqueId();
+    var uniqueId2 = generateUniqueId();
+    
+    assert.notEqual(uniqueId, uniqueId2);
+    // Anything else I should be testing in here?
+});
+
+QUnit.test("emptyCalendarState test", function(assert) {
+    assert.expect(4);
+    var params = {startDate: "2017-02-14" , endDate: "2017-02-20" , calendarTitle: ""};
+    var state = emptyCalendarState(params);
+    
+    assert.equal(state.title, "", "Checking state title");
+    assert.equal(state.startDateString, "20170214", "Checking startDateString");
+    assert.equal(state.endDateString, "20170220", "Checking endDateString");
+    // What kind of assertion for uniqueId?
+    assert.deepEqual(state.checkedDays, {});
+});
+
+// Testing calendar object 
+QUnit.module( "Calendar Tests", {
+
+beforeEach: function() {
+    // Prepare something once for all tests
+    this.params = {startDate: "2017-02-14" , endDate: "2017-02-20" , calendarTitle: "Test Calendar"};
+    //console.log(this.params);
+    this.state = emptyCalendarState(this.params);
+    
+    this.calendar = new Calendar(this.state);
+    
+    
+}
+
+});
+
+QUnit.test("Init calendar object test", function(assert) {
+    assert.expect(7);
+  
+    assert.deepEqual(this.calendar.state, this.state, "Check state is correct");
+    assert.ok(moment.isMoment(this.calendar.startDate, "startDate is a moment obj")); 
+    assert.ok(moment.isMoment(this.calendar.endDate, "endDate is a moment obj"));
+    assert.equal(this.calendar.startDate.format("YYYYMMDD"), "20170214", "Checking startDate object's date");
+    assert.equal(this.calendar.endDate.format("YYYYMMDD"), "20170220", "Checking endDate object's date");
+    assert.equal(this.calendar.monthObjects.length, 1);
+    assert.equal(typeof(this.calendar.monthObjects[0]), "object", "Month object is object");
+});
+
+
+
+QUnit.test("addMonth calendar method test", function(assert) {
+    assert.expect(4);
+    this.calendar.addMonth();
+    
+    assert.equal(this.calendar.endDate.format("YYYYMMDD"), "20170320", "Check one month was added to endDate object");
+    assert.equal(this.calendar.state.endDateString, "20170320", "Check calendar endDateString added a month");
+    assert.equal(this.calendar.monthObjects.length, 2, "Check month was added to monthObjects array");
+    assert.equal(typeof(this.calendar.addMonth()), "object", "addMonth returns an object");
+});
+
+QUnit.test("generateMonthObjects calendar method test", function(assert) {
+    
+    assert.expect(5);
+    
+    var monthObjects = this.calendar.generateMonthObjects(moment("20170101", "YYYYMMDD"), moment("20170214", "YYYYMMDD"));
```

Comment:
You also want to test corner cases and errors. What if the start and end dates are the same? What if they are one day apart? What if the end date is earlier than start date? Does the function return the "right" answer in all these cases?